riposte-core/src/test/java/com/nike/riposte/server/handler/RoutingHandlerTest.java

```diff
@@ -182,4 +196,99 @@ public void findSingleEndpointForExecution_throws_MultipleMatchingEndpointsExcep
+        // given
+        doReturn(null).when(endpointMock).maxRequestSizeInBytesOverride();
+
+        maxRequestSizeInBytes = 10;
+        httpHeaders.set(CONTENT_LENGTH, 100);
+        httpHeaders.set(CONTENT_TYPE, APPLICATION_JSON);
+        handlerSpy = spy(new RoutingHandler(endpoints, maxRequestSizeInBytes));
+
+        // when
+        Throwable thrownException = Assertions.catchThrowable(() -> handlerSpy.doChannelRead(ctxMock, msg));
+
+        // then
+        assertThat(thrownException).isExactlyInstanceOf(TooLongFrameException.class);
+        assertThat(thrownException.getMessage()).isEqualTo("Content-Length header value exceeded configured max request size of 10");
+    }
+
+    @Test
+    public void doChannelRead_HttpRequest_endpoint_overridden_max_request_size_throws_exception() {
+        // given
+        doReturn(100).when(endpointMock).maxRequestSizeInBytesOverride();
+
+        maxRequestSizeInBytes = 99;
+        httpHeaders.set(CONTENT_LENGTH, 101);
+        httpHeaders.set(CONTENT_TYPE, APPLICATION_JSON);
+
+        handlerSpy = spy(new RoutingHandler(endpoints, maxRequestSizeInBytes));
+
+        // when
+        Throwable thrownException = Assertions.catchThrowable(() -> handlerSpy.doChannelRead(ctxMock, msg));
+
+        // then
+        assertThat(thrownException).isExactlyInstanceOf(TooLongFrameException.class);
+        assertThat(thrownException.getMessage()).isEqualTo("Content-Length header value exceeded configured max request size of 100");
+    }
+
+    @Test
+    public void doChannelRead_HttpRequest_under_max_global_request_size_processed_successfully() {
+        // given
+        doReturn(null).when(endpointMock).maxRequestSizeInBytesOverride();
+
+        maxRequestSizeInBytes = 101;
+        httpHeaders.set(CONTENT_LENGTH, 100);
+        httpHeaders.set(CONTENT_TYPE, APPLICATION_JSON);
+        handlerSpy = spy(new RoutingHandler(endpoints, maxRequestSizeInBytes));
+
+        // when
+        PipelineContinuationBehavior result = handlerSpy.doChannelRead(ctxMock, msg);
+
+        // then
+        assertThat(result).isEqualTo(PipelineContinuationBehavior.CONTINUE);
+    }
+
+    @Test
+    public void doChannelRead_HttpRequest_transfer_encoding_chunked_processed() {
+        // given
+        doReturn(null).when(endpointMock).maxRequestSizeInBytesOverride();
+
+        maxRequestSizeInBytes = 101;
+        httpHeaders.set(TRANSFER_ENCODING, CHUNKED);
+        httpHeaders.set(CONTENT_TYPE, APPLICATION_JSON);
+        handlerSpy = spy(new RoutingHandler(endpoints, maxRequestSizeInBytes));
+
+        // when
+        PipelineContinuationBehavior result = handlerSpy.doChannelRead(ctxMock, msg);
+
+        // then
+        assertThat(result).isEqualTo(PipelineContinuationBehavior.CONTINUE);
+    }
+
+    @Test
+    public void doChannelRead_HttpRequest_request_size_validation_endpoint_overridden_processed_message() {
```

Comment:
Yeah I tend to just mix & match the given/when/then bits to come up with the test names. `[target_method/code]_[does_thing_i_want/expect]_[given_some_scenario]`. Thanks for humoring me.